datalad/distribution/tests/test_update.py

```diff
@@ -146,3 +143,135 @@ def test_update_fetch_all(src, remote_1, remote_2):
     ds.repo.get_file_key("first.txt")  # raises if unknown
     eq_([False], ds.repo.file_has_content(["first.txt"]))
 
+
+@with_tempfile(mkdir=True)
+@with_tempfile(mkdir=True)
+def test_newthings_coming_down(originpath, destpath):
+    origin = GitRepo(originpath, create=True)
+    create_tree(originpath, {'load.dat': 'heavy'})
+    Dataset(originpath).add('load.dat')
+    ds = install(source=originpath, path=destpath)
+    assert_is_instance(ds.repo, GitRepo)
+    assert_in('origin', ds.repo.get_remotes())
+    # turn origin into an annex
+    origin = AnnexRepo(originpath, create=True)
+    # clone doesn't know yet
+    assert_false(knows_annex(ds.path))
+    # but after an update it should
+    # no merge, only one sibling, no parameters should be specific enough
+    ds.update()
+    assert(knows_annex(ds.path))
+    # no branches appeared
+    eq_(ds.repo.get_branches(), ['master'])
+    # now merge, and get an annex
+    ds.update(merge=True)
+    assert_in('git-annex', ds.repo.get_branches())
+    assert_is_instance(ds.repo, AnnexRepo)
+    # should be fully functional
+    testfname = opj(ds.path, 'load.dat')
+    assert_false(ds.repo.file_has_content(testfname))
+    ds.get('.')
+    ok_file_has_content(opj(ds.path, 'load.dat'), 'heavy')
```

Comment:
Indeed. Evolution of that test somewhat distorted the original idea. Conveniently, `ds.repo.file_has_content(testfname)` returns False for a file that is guaranteed to have its content with it. So we can leave it as is!